### PR TITLE
Improve benchmarks results output and features

### PR DESCRIPTION
**Pull request by pentschev**
The following improvements are introduced:

- More consistency formatting of results output
- Latency results
- `--n-warmup-iter` argument
- `--no-detailed-report` argument to `send-recv.py`
- Remove Dask dependency on benchmarks when unavailable

### CODE REVIEW

File: benchmarks/send-recv-core.py [hunk 1]
```diff
@@ -293,32 +302,39 @@ def op_completed():
...
             worker.progress()
 
     if args.cuda_profile:
         xp.cuda.profiler.stop()
 
     assert len(times) == args.n_iter
+    bw_avg = format_bytes(2 * args.n_iter * args.n_bytes / sum(times))
+    bw_med = format_bytes(2 * args.n_bytes / np.median(times))
+    lat_avg = int(sum(times) * 1e9 / (2 * args.n_iter))
+    lat_med = int(np.median(times) * 1e9 / 2)
```
Comment (wence-) [inline benchmarks/send-recv-core.py:319]: Could use `dask.utils.format_time` here instead.

File: benchmarks/send-recv-core.py [hunk 2]
```diff
@@ -329,22 +345,23 @@ def op_completed():
...
-        print("--------------------------")
+        print_separator(separator="=")
+        print_key_value(key="Iterations", value="Bandwidth, Latency")
+        print_separator(separator="-")
         for i, t in enumerate(times):
             ts = format_bytes(2 * args.n_bytes / t)
-            ts = (" " * (9 - len(ts))) + ts
-            print("%03d         |%s/s" % (i, ts))
+            lat = int(t * 1e9 / 2)
+            print_key_value(key=i, value=f"{ts}/s, {lat}ns")
```
Comment (wence-) [inline benchmarks/send-recv-core.py:364]: `dask.utils.format_time` again.

**Review by wence- — CHANGES_REQUESTED** [review]
Overall this looks good with the nitpick to do with `format_time`

File: benchmarks/send-recv-core.py (hunk 1 above)
Comment (pentschev) [reply benchmarks/send-recv-core.py:319]: We can't have Dask dependencies in UCX-Py, it is used by projects who do not use Dask.

File: benchmarks/send-recv-core.py (hunk 2 above)
Comment (pentschev) [reply benchmarks/send-recv-core.py:364]: Ditto.

File: benchmarks/send-recv-core.py (hunk 1 above)
Comment (wence-) [reply benchmarks/send-recv-core.py:319]: The `format_bytes` call needs to go then (since that comes from `dask.utils` AFAICS)

Comment (pentschev) [reply benchmarks/send-recv-core.py:319]: You're right, we need to remove that. I'll work on that later today or next week.

Comment (pentschev) [reply benchmarks/send-recv-core.py:315]: This is done now in https://github.com/rapidsai/ucx-py/pull/863/commits/8741b275d2c69cec2b27289bfa819cea2301fdff .

**Review by wence- — APPROVED** [review]

**Comment by wence-** [comment]
Looks good.

**Comment by pentschev** [comment]
Thanks for the review @wence- !

**Review by quasiben — APPROVED** [review]

**Pull request merged**